### PR TITLE
added cnetcheck tests

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

Converts cnetcheck to application and replace tests with gtest 

## Motivation and Context

This test takes up 3817 Mb in the ISIS3TESTDATA area...

## How Has This Been Tested?

This PR is the test. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/cnetcheckFunctionalTests.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+#include "ControlPoint.h"
+
+#include "Fixtures.h"
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/cnetcheck.xml").expanded();
+
+TEST_F(ThreeImageNetwork, FunctionalTestCnetcheckCamera) {
```
Comment (Kelvinrr) [inline isis/tests/cnetcheckFunctionalTests.cpp:26]: probably worth updating these tests to have Functional in the name over app test.

**Comment by acpaquette** [comment]
@Kelvinrr Looks like you're missing the new .cpp and .h files

**Comment by Kelvinrr** [comment]
yeah, I noticed when I looked at Jenkins this AM. Whoops 🤷‍♀

File: isis/tests/cnetcheckFunctionalTests.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  m2->SetCubeSerialNumber(cube2Serial);
+  
+  ControlPoint *newPoint = new ControlPoint(); 
+  newPoint->Add(m1);
+  newPoint->Add(m2);
+  
+  network->AddPoint(newPoint);
+  
+  Pvl log;
+  QString results = cnetcheck(*network, *cubeList, options, &log);
```
Comment (kaitlyndlee) [inline isis/tests/cnetcheckFunctionalTests.cpp]: Small thing, I don't think we use results anywhere in the test.

Comment (Kelvinrr) [reply isis/tests/cnetcheckFunctionalTests.cpp]: Originally I was gonna test against it but it really doesn't have super valuable information worth testing. Probably remove storing the results.

**Comment by Kelvinrr** [comment]
Can we get this reviewed by more people, approved and merged?

**Review by jessemapel — APPROVED** [review]

**Pull request merged**